### PR TITLE
feat: enhance publication `hasCollectedByMe` to replace deprecated `hasOptimisticCollectedByMe`

### PR DESCRIPTION
**Pull request by krzysu**
also move `activeProfileIdentifierVar` to api-bindings package

### CODE REVIEW

File: packages/api-bindings/src/graphql/client.graphql [hunk 1]
```diff
@@ -16,7 +16,7 @@ extend type Profile {
 
 extend type Post {
   """
-  @deprecated this will be replaced by collectState
+  @deprecated use hasCollectedByMe instead
```
Comment (krzysu) [inline packages/api-bindings/src/graphql/client.graphql:19]: this is not added to generated .ts files, so end-user won't see them in their IDE.

File: packages/api-bindings/src/apollo/cache/publicationTypePolicy.ts [hunk 2]
```diff
@@ -76,6 +78,23 @@ const collectPolicy = (
...
 };
 
+const hasCollectedByMe = (existing: boolean, { readField }: FieldFunctionOptions): boolean => {
+  // if collected already then just return, it can't be undone
+  if (existing === true) return existing;
+
+  const publicationId = readField('id') as PublicationId;
+
+  const isCollectTransactionForThisPublication = isCollectTransactionFor({
+    publicationId,
```
[the remark this answers is not in the document]
Comment (cesarenaldi) [reply packages/api-bindings/src/apollo/cache/publicationTypePolicy.ts:88]: Similarly to what we did for `recentTransactionsVar` I think the `activeProfileIdentifierVar` should be moved in `@lens-protocol/api-bindings` and then imported and used by the `ActiveProfilePresenter` and `useActiveProfileIdentifierVar` in the `@lens-protocol/react` package. Once you have the `activeProfileIdentifierVar` in this package you can simply import it and use it here.

[the remark this answers is not in the document]
Comment (krzysu) [reply packages/api-bindings/src/apollo/cache/publicationTypePolicy.ts:92]: done, I guess it's ready for review now

**Review by cesarenaldi — CHANGES_REQUESTED** [review]
Neat, can you please add a changeset?

File: .changeset/large-mugs-fetch.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@lens-protocol/api-bindings": minor
+"@lens-protocol/react": minor
```
Comment (cesarenaldi) [inline .changeset/large-mugs-fetch.md:3]: Sorry, one last thing, can you please add `@lens-protocol/react-web` here? We need to update the deps with the updated `@lens-protocol/react`.
```suggestion
"@lens-protocol/react": minor
"@lens-protocol/react-web": minor
```

**Pull request merged**